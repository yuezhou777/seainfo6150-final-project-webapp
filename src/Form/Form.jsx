import React from "react";
import { useHistory } from "react-router-dom";
import styles from "../Form/Form.module.css";

const Form = () => {

  const history = useHistory();

  const submitHandler = (e) => {
    e.preventDefault();
    history.push("/thankyou");
  };

  return (
    <div className={styles.container}>
      <form onSubmit={submitHandler}>
        <fieldset className={styles.formContent}>
          <h2>Adoption Form</h2>
          <div>
            <label htmlFor="firstName">First Name : </label>
            <input type="text" name="firstName" id="firstName"
                  required
            />
          </div>
          <div>
            <label htmlFor="lastName">Last Name : </label>
            <input type="text" name="lastName" id="lastName"
                  required
            />
          </div>
          <div>
            <label htmlFor="email">Email : </label>
            <input type="text" name="email" id="email"
                  pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$"
                  placeholder="abc@gmail.com"
                  required
            />
          </div>
          <div>
            <label htmlFor="phoneNumber">Phone Number : </label>
            <input type="text" name="phoneNumber" id="phonephoneNumberNum"
                  pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                  placeholder="xxx-xxx-xxxx"
            />
          </div>
          <div>
            <input className={styles.button} type="submit" value="Register"></input>
          </div>
        </fieldset>
      </form>
    </div>
  )
}

export default Form
