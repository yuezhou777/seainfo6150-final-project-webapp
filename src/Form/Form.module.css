.container {
    display: flex;
    flex-flow: column;
}

.formContent {
    line-height: 40px;
}

.button {
    background-color: orange;
}

.button:hover {
    background-color: greenyellow;
}