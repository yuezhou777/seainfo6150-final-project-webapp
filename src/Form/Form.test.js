import React from 'react'
import {render} from "@testing-library/react";
import Form from "./Form.jsx";

describe('form test', () => {

    it('render Form successfully ', () => {
        const {container} = render(<Form />);

        expect(container).toMatchSnapshot();
    });

})