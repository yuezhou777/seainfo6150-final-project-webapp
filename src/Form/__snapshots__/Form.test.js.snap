// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form test render Form successfully  1`] = `
<div>
  <div
    class="container"
  >
    <form>
      <fieldset
        class="formContent"
      >
        <h2>
          Adoption Form
        </h2>
        <div>
          <label
            for="firstName"
          >
            First Name : 
          </label>
          <input
            id="firstName"
            name="firstName"
            required=""
            type="text"
          />
        </div>
        <div>
          <label
            for="lastName"
          >
            Last Name : 
          </label>
          <input
            id="lastName"
            name="lastName"
            required=""
            type="text"
          />
        </div>
        <div>
          <label
            for="email"
          >
            Email : 
          </label>
          <input
            id="email"
            name="email"
            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,}$"
            placeholder="abc@gmail.com"
            required=""
            type="text"
          />
        </div>
        <div>
          <label
            for="phoneNumber"
          >
            Phone Number : 
          </label>
          <input
            id="phonephoneNumberNum"
            name="phoneNumber"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            placeholder="xxx-xxx-xxxx"
            type="text"
          />
        </div>
        <div>
          <input
            class="button"
            type="submit"
            value="Submit"
          />
        </div>
      </fieldset>
    </form>
  </div>
</div>
`;
